sui_core/src/authority.rs

```diff
@@ -517,12 +518,38 @@ impl AuthorityState {
             return Err(SuiError::TooManyItemsError(MAX_ITEMS_LIMIT));
         }
 
-        let (batches, transactions) = self
+        let (mut batches, transactions): (
+            Vec<SignedBatch>,
+            Vec<(TxSequenceNumber, TransactionDigest)>,
+        ) = self
             ._database
             .batches_and_transactions(request.start, request.end)?;
 
+        // Append the transaction info to individual transactions if requested
+        let mut transactions_with_info = Vec::new();
+        for (seq, digest) in transactions.iter() {
+            if request.include_tx_info {
+                let tx_info = self.make_transaction_info(digest).await?;
+                transactions_with_info.push((*seq, *digest, Some(Box::new(tx_info))));
```

Comment:
Hm, I am not a great fan of making the authority do the work of expanding the digests into full trasnaction infos, when perfectly good client side APIs exist (and can use other stores) to get it. However, lets not block on this now, we can refactor down the line. It is not incorrect or anything.

Reply:
@gdanezis if the authority is not going to send the full thing, then what do you suggest?  Each follower / replica / full node ask the authority for a large number of full transactions?   I was hoping for this to enable the event subscriptions...   maybe I'll ping you offline.

If the concern is about offloading the authority, that can only be solved by a sidecar which handles read requests.... either that sidecar is a full node, or what Ade and I have proposed which is offloaded storage which can handle many more reads....